ygen/protogen_test.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package ygen
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/kylelemons/godebug/pretty"
+	"github.com/openconfig/goyang/pkg/yang"
+)
+
+func protoMsgEq(a, b protoMsg) bool {
+	if a.Name != b.Name {
+		return false
+	}
+
+	if a.YANGPath != b.YANGPath {
+		return false
+	}
+
+	// Avoid flakes by comparing the fields in an unordered data structure.
+	fieldMap := func(s []*protoMsgField) map[string]*protoMsgField {
+		e := map[string]*protoMsgField{}
+		for _, m := range s {
+			e[m.Name] = m
+		}
+		return e
+	}
+
+	if !reflect.DeepEqual(fieldMap(a.Fields), fieldMap(b.Fields)) {
+		return false
+	}
+
+	return true
+}
+
+func TestGenProtoMsg(t *testing.T) {
+	tests := []struct {
+		name                string
+		inMsg               *yangStruct
+		inMsgs              map[string]*yangStruct
+		inUniqueStructNames map[string]string
+		wantMsg             protoMsg
+		wantErr             bool
+	}{{
+		name: "simple message with only scalar fields",
+		inMsg: &yangStruct{
+			name: "MessageName",
+			entry: &yang.Entry{
+				Name: "message-name",
+				Dir:  map[string]*yang.Entry{},
+			},
+			fields: map[string]*yang.Entry{
+				"field-one": {
+					Name: "field-one",
```

Comment:
test unsafe field char replacement?

Reply:
This actually does test the field character replacement -- since "-" is allowed in YANG and not in protobuf. I cleaned this up since there was actually a case that could never occur.